pilot/pkg/config/kube/crdclient/cache_handler.go

```diff
@@ -82,16 +82,11 @@ func createCacheHandler(cl *Client, schema collection.Schema, i informers.Generi
 	h := &cacheHandler{
 		client:   cl,
 		schema:   schema,
-		informer: i.Informer(),
-	}
-	h.lister = func(namespace string) cache.GenericNamespaceLister {
-		if schema.Resource().IsClusterScoped() {
-			return i.Lister()
-		}
-		return i.Lister().ByNamespace(namespace)
+		informer: informer.NewFilteredSharedIndexInformer(cl.namespacesFilter, i.Informer()),
 	}
```

Comment:
`lister: cache.NewGenericLister(h.informer.GetIndexer(), schema.Resource().GroupVersionResource().GroupResource())`

can we do this?

Reply:
Definitely we can implement it in theory, but I think currently it is enough to use

NewGenericLister does not provide much benefit to our case

Reply:
As you can see with generic lister for clusterscoped resource, we need to handle differently. however with underlying indexer, we do not need to care about it, that is why I believe this way is more simple

Reply:
Actually, all the indexer related code are in pkg/kube, the fiteredInformer does only provide Get List interface.

Reply:
I am not sure how it's simpler. The lister requires 1 single if statement here - the indexer approach has like 50 lines of added complexity throughout the PR of more complex error handling, type casting, and still has the if for namespaces but at each List call instead of in one single place 

Listers are the Kubernetes standard for his to list objects, I don't see a compelling reason to move away from that. It also seems unrelated to the rest of the PR.

Reply:
GenericLister returns runtime.Object, which also need type cast, error handling. I do not have strong preference though

Reply:
yes but we don't need this: https://github.com/istio/istio/pull/36639/files#diff-eff3d408ac506e9395b51f480c999a588f9fa408e392ce77506f25ceae00d6a1R328


and we now have to do:
```
	item, _, err := h.informer.GetIndexer().GetByKey(KeyFunc(namespace, name))
	if item == nil || err != nil {
```

instead of just 

```
	obj, err := h.lister(namespace).Get(name)
	if err != nil {
```